### PR TITLE
refactor(api, hardware): ot3: Handle move position response

### PR DESCRIPTION
**Pull request by sfoster1**
After https://github.com/Opentrons/ot3-firmware/pull/304 , the firmware will return the current value of the position tracker in micrometers rather than steps, which means we can use its value to estimate the current position of the system. Add handling for the position value in MoveCompleted messages from Canbus and bubble it up to the position tracker in the controller.

### CODE REVIEW

**Review by amitlissack — APPROVED** [review]
Looks nice to me

File: hardware/tests/firmware_integration/test_move_groups.py [hunk 1]
```diff
@@ -80,3 +83,101 @@ async def test_add_moves(
...
+            },
+            acceleration={motor_node: 0 for motor_node in all_motor_nodes},
+            duration=np.float64(1),
+            present_nodes=all_motor_nodes,
+        )
+    ]
+    prep_runner = MoveGroupRunner([prep_move])
+    # throw away this position - it may change depending on previous state of
+    # simulators
+    await prep_runner.run(can_messenger)
```
Comment (DerekMaggio) [inline hardware/tests/firmware_integration/test_move_groups.py:114]: Are you running this before home to make sure the robot is a state other than home before testing homing?

File: hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py [hunk 2]
```diff
@@ -479,6 +483,8 @@ async def test_single_move(
             )
         ]
     )
+    assert len(position) == 1
+    assert position[0][1].payload.current_position_um.value == 246000
```
Comment (DerekMaggio) [inline hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py:487]: Threw me off for a second why we were expecting it to be 246000. 
Realized that it was the distance converted from mm to um. 
Pulling these scalars out to constants such as `EXPECTED_HEAD_MICROMETERS` would make this instantly clear

**Review by DerekMaggio — APPROVED** [review]
Looks good to me. Just a question and a suggestion

File: hardware/tests/firmware_integration/test_move_groups.py (hunk 1 above)
Comment (sfoster1) [reply hardware/tests/firmware_integration/test_move_groups.py:114]: Yeah

File: hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py (hunk 2 above)
Comment (sfoster1) [reply hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py:487]: that's what the `um` is though. i'm usually all for putting constants in variables with names, but in the case of literal comparisons in unit tests I personally get way more value out of seeing the number right there.

File: hardware/tests/firmware_integration/test_move_groups.py (hunk 1 above)
Comment (sfoster1) [reply hardware/tests/firmware_integration/test_move_groups.py:114]: Yeah, that's correct. We have to home anyway so we might as well simultaneously test it

File: hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py (hunk 2 above)
Comment (sfoster1) [reply hardware/tests/opentrons_hardware/hardware_control/test_move_group_runner.py:487]: I guess... I like named values for constants in code, but when doing immediate comparisons in tests I'm a much bigger fan of just having the number there to look at. Plus the unit's in the left hand variable name!

**Comment by caila-marashaj** [comment]
Tested homing and move commands, looks good to me!

**Review by caila-marashaj — APPROVED** [review]

**Pull request merged**